src/str/transform.php

```diff
@@ -242,9 +242,89 @@ function replace_every_ci(
   );
 }
 
-function reverse(
-  string $string,
+/**
+ * Returns the "haystack" string with all occurrences of the keys of
+ * `$replacements` replaced by the corresponding values.
+ *
+ * Once a substring has been replaced, its new value will not be searched
+ * again.
+ *
+ * - For having new values searched again, see `Str\replace_every()`.
+ */
+<<__Rx>>
+function replace_every_nonrecursive(
+  string $haystack,
+  KeyedContainer<string, string> $replacements,
+): string {
+  invariant(
+    !C\contains_key($replacements, ''),
+    'Expected non-empty keys only.',
+  );
+
+  /* HH_FIXME[2049] calling stdlib directly */
+  /* HH_FIXME[4107] calling stdlib directly */
+  /* HH_FIXME[4200] Rx calling non-rx */
+  return \strtr($haystack, $replacements);
+}
+
+/**
+ * Returns the "haystack" string with all occurrences of the keys of
+ * `$replacements` replaced by the corresponding values (case-insensitive).
+ *
+ * Once a substring has been replaced, its new value will not be searched
+ * again.
+ *
+ * - For having new values searched again, see `Str\replace_every_ci()`.
```

Comment:
Could you also update the `replace_every[_ci]` docblock similarly, to mention the nonrecursive version?